homework4.txt

```diff
@@ -0,0 +1,261 @@
+
+3)List all ESTABLISHED TCP connections with lsof (only them, without any other commands in a pipe)
+
+MacBook-Air-User:~ user$ lsof -i 6
+COMMAND    PID USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
+rapportd 94728 user    4u  IPv6 0x39374009053828c1      0t0  TCP *:56103 (LISTEN)
+
+/*
+Виртуальная машина ничего не выводит, поскольку нет соединений
+*/
+
+2. Systemd:
+
+1) Write two daemons one should be a simple daemon and do "sleep 10" after a start and then do "echo 1 > /tmp/homework", the second one should be oneshot and do "echo 2 > /tmp/homework" without any sleep
+
+[dmitrii@localhost ~]$ cd /tmp
+[dmitrii@localhost tmp]$ ls -l
+итого 0
+drwx------. 3 root    root    17 окт 30 09:01 systemd-private-05e02df8dc924dea8e27d4eed9b1cf38-chronyd.service-Wvt7zi
+drwx------. 2 dmitrii dmitrii 21 окт 28 10:58 tmux-1000
+[dmitrii@localhost tmp]$ touch homework
+[dmitrii@localhost tmp]$ ls -l
+итого 0
+-rw-rw-r--. 1 dmitrii dmitrii  0 ноя  9 02:40 homework
+drwx------. 3 root    root    17 окт 30 09:01 systemd-private-05e02df8dc924dea8e27d4eed9b1cf38-chronyd.service-Wvt7zi
+drwx------. 2 dmitrii dmitrii 21 окт 28 10:58 tmux-1000
+
+[dmitrii@localhost ~]$ vim daemon1.sh
+[dmitrii@localhost ~]$ chmod u+x daemon1.sh 
+[dmitrii@localhost ~]$ ./daemon1.sh 
+[dmitrii@localhost ~]$ cd /tmp
+[dmitrii@localhost tmp]$ cat homework
+1
+
+/*
+daemon1.sh
+
+#! /bin/bash
+
+sleep 10
+echo "1" > /tmp/homework
+*/
+
+[dmitrii@localhost ~]$ vim daemon2.sh
+[dmitrii@localhost ~]$ chmod u+x daemon2.sh
+[dmitrii@localhost ~]$ ./daemon2.sh 
+[dmitrii@localhost ~]$ cat /tmp/homework
+2
+
+2) Make the second depended on the first one (should start only after the first)
+
+/*
+daemon1.sh
+
+#! /bin/bash
+
+sleep 10
+echo "1" > /tmp/homework
+./daemon2.sh
+
+*/
+
+3) Write a timer for the second one and configure it to run on 01.01.2019 at 00:00
+
+[root@localhost tmp]# vim daemon2.sh
+[root@localhost tmp]# chmod +x daemon2.sh
+
+/*
+homework.timer
+
+[Unit]
+Description=Homework
+
+[Timer]
+OnCalendar=01-01-19 0:0:0
+Unit=homework.service
+
+*/
+
+/*
+homework.service
+
+[Unit]
+Description=Homework
+
+[Service]
+Type=oneshot
+ExecStart=/tmp/daemon2.sh
+*/
+
+[root@localhost system]# ls -l
+итого 20
+drwxr-xr-x. 2 root root   57 окт 23 19:08 basic.target.wants
+lrwxrwxrwx. 1 root root   41 окт 23 19:07 dbus-org.fedoraproject.FirewallD1.service -> /usr/lib/systemd/system/firewalld.service
+lrwxrwxrwx. 1 root root   46 окт 23 19:07 dbus-org.freedesktop.NetworkManager.service -> /usr/lib/systemd/system/NetworkManager.service
+lrwxrwxrwx. 1 root root   57 окт 23 19:07 dbus-org.freedesktop.nm-dispatcher.service -> /usr/lib/systemd/system/NetworkManager-dispatcher.service
+lrwxrwxrwx. 1 root root   37 окт 23 19:12 default.target -> /lib/systemd/system/multi-user.target
+drwxr-xr-x. 2 root root   87 окт 23 19:07 default.target.wants
+drwxr-xr-x. 2 root root   32 окт 23 19:07 getty.target.wants
+-rw-r--r--. 1 root root   78 ноя  9 06:17 homework.service
```

Comment:
You have to create two daemons, not only one, and one of them has to depend on another.